### PR TITLE
Add the `sv_between()` rule fn

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `sv_between()` rule fn and while it works just fine, this'll be a work in progress. If you don't mind I'd like to discuss some ideas for developing this further (I'll create an issue for this).

### CODE REVIEW

File: R/rules.R [hunk 1]
```diff
@@ -210,3 +210,80 @@ sv_integer <- function(message = "An integer is required",
...
+  # TODO: normalize lang (ensures that NULL -> "en" and other inputs
+  # match supported langs in shinyvalidate)
+  # lang <- normalize_lang(lang)
+  
+  # TODO: The gluestring will be obtained via a function that obtains
+  # a localized string vector and focuses it on the spoken language
+  # TODO: Use a safe version of `glue::glue()`
+  # message <- glue::glue(get_lsv("between")[[lang]])
+  if (is.null(message)) {
+    message <- glue::glue("The field must be between {minimum} and {maximum}")
```
Comment (jcheng5) [inline R/rules.R:270]: Let's use `glue_data` and safe transformer

File: R/rules.R [hunk 2]
```diff
@@ -210,3 +210,80 @@ sv_integer <- function(message = "An integer is required",
...
+                       maximum,
+                       allowNA = FALSE,
+                       allowNaN = FALSE,
+                       message = NULL,
+                       lang = NULL) {
+  force(allowNA)
+  force(allowNaN)
+  force(message)
+  
+  # TODO: consider adding arg for specifying inclusive bounds
```
Comment (jcheng5) [inline R/rules.R:259]: Definitely need this and I think default should be inclusive

File: R/rules.R [hunk 3]
```diff
@@ -210,3 +210,80 @@ sv_integer <- function(message = "An integer is required",
...
+      return(message)
+    }
+    # TODO: use the "noNaN" lsv for the message
+    if (!allowNaN && any(is.nan(value))) {
+      return(message)
+    }
+    
+    # TODO: ensure that `value` is coercible to numeric
+    
+    if ((as.numeric(value) < minimum) | (as.numeric(value) > maximum)) {
```
Comment (jcheng5) [inline R/rules.R:285]: Don't cast

File: R/rules.R (hunk 1 above)
Comment (rich-iannone) [reply R/rules.R:270]: Now fixed.

File: R/rules.R (hunk 3 above)
Comment (rich-iannone) [reply R/rules.R:285]: Now updated to remove the `as.numeric()`s.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**